library/src/main/java/com/datatorrent/lib/io/fs/FileMerger.java

```diff
@@ -0,0 +1,161 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.datatorrent.lib.io.fs;
+
+import java.io.IOException;
+import java.io.OutputStream;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import org.apache.hadoop.fs.Path;
+
+import com.datatorrent.api.AutoMetric;
+import com.datatorrent.api.Context;
+import com.datatorrent.api.Context.OperatorContext;
+import com.datatorrent.lib.io.fs.Synchronizer.OutputFileMetadata;
+
+/**
+ * This operator merges the blocks into a file. The list of blocks is obtained
+ * from the OutputFileMetadata. The implementation extends FileStitcher (which
+ * uses reconciler), hence the file merging operation is carried out in a
+ * separate thread.
+ *
+ */
+public class FileMerger extends FileStitcher<OutputFileMetadata>
+{
+  /**
+   * Flag to control if existing file with same name should be overwritten
+   */
+  private boolean overwriteOnConflict = true;
+
+  private static final Logger LOG = LoggerFactory.getLogger(FileMerger.class);
+
+  @AutoMetric
+  private long bytesWrittenPerSec;
+
+  private long bytesWritten;
+  private double windowTimeSec;
+
+  @Override
+  public void setup(OperatorContext context)
+  {
+    super.setup(context);
+    windowTimeSec = (context.getValue(Context.OperatorContext.APPLICATION_WINDOW_COUNT)
+        * context.getValue(Context.DAGContext.STREAMING_WINDOW_SIZE_MILLIS) * 1.0) / 1000.0;
+  }
+
+  @Override
+  public void beginWindow(long windowId)
+  {
+    super.beginWindow(windowId);
+    bytesWrittenPerSec = 0;
+    bytesWritten = 0;
+  }
+
+  /* 
```

Comment:
If you don't provide javadoc here, it would pick from parent class, that should be acceptable unless you want to write something different.